SEED-platform/seed
# Apply unit conversion to extra data

#### Any background context you want to provide?
There are 4 different data types outside the normal float, int, string... These are eui, area, ghg, and ghg_intensity. PropertyState fields associated with these four types are given a pint unit and saved in the db as <Quantity (value, 'unit')>. Before displaying values on the Inventory List table the quantity values are converted based on org display unit settings.

Extra data fields are not given a pint unit and are saved in whatever format they were imported as (string or number). Because they don't have pint units they are ignored when the unit conversion takes place allowing data to be misrepresented.

#### What's this PR do?
Adds a pint unit to extra data columns that have been assigned a data type in column settings.

#### How should this be manually tested?
- upload property data with both canonical area columns and extra data area columns (or eui, ghg, ghg_intensity)
- on column settings change the data type of the extra area field to area
- on org settings, change the default display units of area from ft2 to m2
- ensure inventory list values are updated (this should also be visible on the mapping page of the import process)

#### What are the relevant tickets?
#4494

#### Screenshots (if appropriate)
Inventory List
![Screenshot 2024-01-19 at 2 54 58 PM](https://github.com/SEED-platform/seed/assets/58446472/9012f1b5-7803-47b1-bdc7-0f9bde9f792f)
![Screenshot 2024-01-19 at 2 54 37 PM](https://github.com/SEED-platform/seed/assets/58446472/1cb199e3-9abd-4a5e-80f7-3f52eb4d4fa1)

Mapping Screen
![Screenshot 2024-01-19 at 3 39 06 PM](https://github.com/SEED-platform/seed/assets/58446472/01d969f9-1013-4a7e-88e4-18c08d1995f9)
![Screenshot 2024-01-19 at 3 38 11 PM](https://github.com/SEED-platform/seed/assets/58446472/d4ecc36c-0658-4d7b-9e9f-1fe5f486d965)
